api_guard/dist/types/operators/index.d.ts

```diff
@@ -169,7 +169,7 @@ export declare function merge<T, T2, T3, T4, T5, T6>(v2: ObservableInput<T2>, v3
 export declare function merge<T>(...observables: Array<ObservableInput<T> | SchedulerLike | number>): MonoTypeOperatorFunction<T>;
 export declare function merge<T, R>(...observables: Array<ObservableInput<any> | SchedulerLike | number>): OperatorFunction<T, R>;
 
-export declare function mergeAll<T>(concurrent?: number): OperatorFunction<ObservableInput<T>, T>;
+export declare function mergeAll<O extends ObservableInput<any>>(concurrent?: number): OperatorFunction<O, ObservedValueOf<O>>;
```

Comment:
Does this do the correct tuple thing?

Reply:
Well, there's no tuples needed in the output... so it gets the proper union type out of the other side, yeah.